### ISSUE TITLE
async parsing rules are broken

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#7232]
**Tell us about your environment**

* **ESLint Version:**
3.6.0
* **Node Version:**
6.6
* **npm Version:**
whatever it comes with

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`echo 'var foo = { async: 3 };' | eslint --no-eslintrc --stdin --parser-options='ecmaVersion:2017'`

**What did you do? Please include the actual source code causing the issue.**
`echo 'var foo = { async: 3 };' | eslint --no-eslintrc --stdin --parser-options='ecmaVersion:2017'`

**What did you expect to happen?**
No error.

**What actually happened? Please include the actual, raw output from ESLint.**
```sh

<text>
  1:18  error  Parsing error: Unexpected token :

✖ 1 problem (1 error, 0 warnings)
```

`async` is not a keyword, so it should be allowed anywhere.

**Comment by platinumazure** [comment]
Probably an issue in espree.

Just so we can understand impact, does this happen with ecmaVersion of 7/2016?

**Comment by ljharb** [comment]
Nope! just with 8/2017.

**Comment by platinumazure** [comment]
Trying to repro, got some funkiness in my local environment right now. Added "blocked" label due to presumed upstream (espree) issue. Will confirm and mark "accepted" ASAP.

**Comment by mysticatea** [comment]
@ljharb What is the version of Acorn you are using? I think it has been fixed on 4.0.3: https://github.com/ternjs/acorn/pull/470

**Comment by platinumazure** [comment]
Acorn parses this fine and espree reproduces the issue when used directly, so I think it's definitely an espree issue. Unfortunately, that's all I have time for at this point, so someone else feel free to jump on it.

@mysticatea I reproduced on espree master but had no trouble when using espree's acorn dependency (and I just cloned espree and `npm install`'d five minutes ago), so I'm pretty sure there's an espree problem.

**Comment by mysticatea** [comment]
Ah, I'm sorry.
Yes, espree overrides the `parseObj` method completely. So espree needs to be fixed.

**Comment by mysticatea** [comment]
I sent a PR to fix.

**Comment by herodrigues** [comment]
Same here: 3 problems only where async is present.
